rotkehlchen/rotkehlchen.py

```diff
@@ -797,7 +797,8 @@ def query_balances(
                 force_token_detection=ignore_cache,
                 ignore_cache=ignore_cache,
             )
-            balances[str(Location.BLOCKCHAIN)] = blockchain_result.totals.assets
+            if len(blockchain_result.totals.assets):
```

Comment:
explicit. Never compare 0 to a Falsy value. This is a source of many many bugs.
```suggestion
            if len(blockchain_result.totals.assets) != 0:
```